### ISSUE TITLE
Lint-staged integration doesn't stop on failure

### ISSUE THREAD

**Issue by k2snowman69** [snowcoders/sortier#353]
# Language
All

# Sample input source code
A file of which fails to be parsed due to invalid javascript

# Expected output
When running `git commit -a -m "asdfasdf"` that the file which failed to be parsed to stop lint-staged from continuing 

# Actual output (or error message)
lint-stage just continues and commits the unsorted file

**Comment by k2snowman69** [comment]
Well it looks like returning -1 and outputting to the console output is definitely not enough to get lint-staged to exit prematurely so time to look into errors.

**Comment by k2snowman69** [comment]
Yep, looks like we need to throw an exception annoyingly....

**Comment by k2snowman69** [comment]
This will require a major build update. Luckily, we're aiming for 3.0.0 soon so we'll lump this in there :-)

**Comment by k2snowman69** [comment]
Investigating what other tools do so we can be consistent.
### Setup
I created a file called `./src/language.ts` and it contains the following invalid typescript
```
what would you do if you wanted to fail?
```

### Tslint
Results in no errors along with a return code of 0
```
D:\Projects\sortier\cli>npm run tslint -- -c tslint.json --fix ./src/language.ts

> @snowcoders/sortier@2.6.2 tslint D:\Projects\sortier\cli
> tslint "-c" "tslint.json" "--fix" "./src/language.ts"


D:\Projects\sortier\cli>echo Exit Code is %errorlevel%
Exit Code is 0

D:\Projects\sortier\cli>
```

### Prettier
Prettier did throw an error and also returned a return code of 2. This also seems to make lint-stage fail.
```
D:\Projects\sortier\cli>npm run prettier -- --write ./src/language.ts

> @snowcoders/sortier@2.6.2 prettier D:\Projects\sortier\cli
> prettier "--write" "./src/language.ts"

src\language.ts
[error] src\language.ts: SyntaxError: ';' expected. (12:6)
[error]   10 | }
[error]   11 |
[error] > 12 | what would you do if you wanted to fail?
[error]      |      ^
[error]   13 |
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @snowcoders/sortier@2.6.2 prettier: `prettier "--write" "./src/language.ts"`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @snowcoders/sortier@2.6.2 prettier script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\RickyPatel\AppData\Roaming\npm-cache\_logs\2019-07-20T17_51_43_467Z-debug.log

D:\Projects\sortier\cli>echo Exit Code is %errorlevel%
Exit Code is 2
```

**Comment by k2snowman69** [comment]
Investigating a bit more, we have two types of errors it seems... 

### Unsupported files
There are ones where files aren't supported... These make no sense to fire an error on because there's nothing the consumer can do
Current: We log a message but return exit return of 0
```
D:\Projects\sortier\cli>npm run run -- "./!(node_modules)/**/*"

> @snowcoders/sortier@2.6.2 run D:\Projects\sortier\cli
> node bin/index.js "./!(node_modules)/**/*"


Sorting coverage/sort-arrow-sprite.png has failed!
If this is an issue with sortier please provide an issue in Github with minimal source code to reproduce the issue
Error: Could not find language support for file - coverage/sort-arrow-sprite.png
```

### Sortier failed because of an internal error
These are internal errors that sortier can't handle but the source code maybe valid
Current: We log a message but return exit return of 0
```
Sorting dist/reprinter/index.d.ts has failed!
If this is an issue with sortier please provide an issue in Github with minimal source code to reproduce the issue
Error: Unexpected Exception - Node received is null
```

### Sortier failed because file couldn't be compiled
These are scenarios where the file couldn't be sorted because the source is invalid.
Current: We just continue
```
D:\Projects\sortier\cli>npm run run -- "./src/**/*"

> @snowcoders/sortier@2.6.2 run D:\Projects\sortier\cli
> node bin/index.js "./src/**/*"


D:\Projects\sortier\cli>
```

I think the expected behavior should be:
 - Unsupported files - Should log during diagnostic mode, not throw an error and exit code is 0
 - Sortier failed because of an internal error - Should log during normal mode but not throw an error and exit code is 0
 - Sortier failed because file couldn't be compiled - Should log during normal mode and throw an error if possible with exit code !0

That latter one should really be caught by people's builds to be honest but we shouldn't fail silently

**Comment by k2snowman69** [comment]
Unsupported files should log in diagnostic mode - https://github.com/snowcoders/sortier/pull/579

**Comment by k2snowman69** [comment]
Released with 3.0.0
